doc/builders/fetchers.chapter.md

```diff
@@ -82,4 +82,11 @@ This is used with repo.or.cz repositories. The arguments expected are very simil
 
 ## `fetchFromSourcehut` {#fetchfromsourcehut}
 
-This is used with sourcehut repositories. The arguments expected are very similar to fetchFromGitHub above. Don't forget the tilde (~) in front of the user name!
+This is used with sourcehut repositories. Similar to `fetchFromGitHub` above,
+it expects `owner`, `repo`, `rev` and `sha256`, but don't forget the tilde (~)
+in front of the username! Expected arguments also include `vc` ("git" (default)
+or "hg"), `domain` and `recursive`.
+
+If `recursive` is `true`, `fetchFromSourcehut` uses `fetchgit` or `fetchhg`
+with `fetchSubmodules` or `fetchSubrepos` set to `true`, respectively.
```

Comment:
I don't like that the attribute is named differently compared to fetchFromGitHub and fetchFromGitLab.

Reply:
GitHub and GitLab only support git, while SourceHut also supports mercurial and the two VCS use different terminologies (submodule and subrepository, respectively).  If `fetchRecursively` or alike draws better similarity, I'm happy to rename it.

Reply:
but most people will use it to fetch git things and don't even know it supports hg. For example if projects move to sourcehut there will be confusion when the fetcher is changed and the question comes up why that isn't accepted, too.

Reply:
IMHO people should be consulting the documentation instead of guessing which parameter a function would take.  If such reasoning does not change you mind and no one else has an opinion on this, I'm fine with renaming it to `fetchSubmodules` though.

Reply:
Looking at Sourcehut repos (featured and last active) I could only see git repos.
So from a practical PoV `fetchSubmodules` would feel "natural".

It may feel awkward for a Mercurial repo but as long as it's documented it should be usable.

Reply:
Alrighty then, that makes it 2 over 1!